modules/sc-mesh-secure-deployment/src/nats/tests/test_mdm_agent.py

```diff
@@ -96,8 +370,518 @@ def test_validate_response_returns_ok_for_200_status_and_empty_payload(self):
         """
         mock_response = MagicMock()
         mock_response.status_code = 200
-        mock_response.text = '{"payload": {"radios": []}}'
+        mock_response.text = '{"payload": {"radios": [{}]}}'
 
-        result = self.mdm_agent._MdmAgent__validate_response(mock_response,
-                                                             mdm_agent.ConfigType.MESH_CONFIG)
-        self.assertEqual(result, 'FAIL')
+        result = self.agent._MdmAgent__validate_response(mock_response,
+                                                         mdm_agent.ConfigType.MESH_CONFIG)
+        self.assertEqual(result, 'OK')
+
+        mock_response.text = '{"payload": {"features": [{}]}}'
+        result = self.agent._MdmAgent__validate_response(mock_response,
+                                                         mdm_agent.ConfigType.FEATURES)
+        self.assertEqual(result, 'OK')
+
+        mock_response.text = '{"payload": {"debug_config": [{}]}}'
+        result = self.agent._MdmAgent__validate_response(mock_response,
+                                                         mdm_agent.ConfigType.DEBUG_CONFIG)
+        self.assertEqual(result, 'OK')
```

Comment:
I'd say there's also the need to have tests (in a different method) where it tests that `mock_response.text = '{"payload": {"debug_config": [{}]}}'` but with the incorrect config type :bubble_tea:

Reply:
We have received configuration validation in other files.  This is just testing the executor logic that we are really trying to download and empty received config won't cause a re-download.

Reply:
ok, but then there could be some additional tests for what I suggested elsewhere (eventually) :garlic:

Reply:
I took few "seconds" to understand your point... will check.

Reply:
[85ed8fb](https://github.com/tiiuae/mesh_com/pull/460/commits/85ed8fbf4b0fa890ee04be23bab6e6da7a441229)